devpi fails to build with kernel 4.9.13 and GCC 6.3.1

I get these errors:

```
make -C /lib/modules/4.9.13-201.fc25.x86_64/build M=/home/evan/code/devpi modules
make[1]: Entering directory '/usr/src/kernels/4.9.13-201.fc25.x86_64'
  CC [M]  /home/evan/code/devpi/devpi.o
/home/evan/code/devpi/devpi.c:44:26: error: unknown type name ‘ctl_table’
 static int proc_dopimode(ctl_table* table, int write, void __user* buffer, size_t* lenp, loff_t* ppos)
                          ^~~~~~~~~
/home/evan/code/devpi/devpi.c:78:8: error: unknown type name ‘ctl_table’
 static ctl_table pi_table[] = {
        ^~~~~~~~~
/home/evan/code/devpi/devpi.c:79:5: warning: braces around scalar initializer
     {
     ^
/home/evan/code/devpi/devpi.c:79:5: note: (near initialization for ‘pi_table[0]’)
/home/evan/code/devpi/devpi.c:80:9: error: field name not in record or union initializer
         .procname = "max_opened",
         ^
/home/evan/code/devpi/devpi.c:80:9: note: (near initialization for ‘pi_table[0]’)
/home/evan/code/devpi/devpi.c:80:21: warning: initialization makes integer from pointer without a cast [-Wint-conversion]
         .procname = "max_opened",
                     ^~~~~~~~~~~~
/home/evan/code/devpi/devpi.c:80:21: note: (near initialization for ‘pi_table[0]’)
/home/evan/code/devpi/devpi.c:80:21: error: initializer element is not computable at load time
/home/evan/code/devpi/devpi.c:80:21: note: (near initialization for ‘pi_table[0]’)
/home/evan/code/devpi/devpi.c:81:9: error: field name not in record or union initializer
         .data = &max_opened,
         ^
/home/evan/code/devpi/devpi.c:81:9: note: (near initialization for ‘pi_table[0]’)
/home/evan/code/devpi/devpi.c:81:17: warning: excess elements in scalar initializer
         .data = &max_opened,
                 ^
/home/evan/code/devpi/devpi.c:81:17: note: (near initialization for ‘pi_table[0]’)
/home/evan/code/devpi/devpi.c:82:9: error: field name not in record or union initializer
         .maxlen = sizeof(unsigned),
         ^
/home/evan/code/devpi/devpi.c:82:9: note: (near initialization for ‘pi_table[0]’)
/home/evan/code/devpi/devpi.c:82:19: warning: excess elements in scalar initializer
         .maxlen = sizeof(unsigned),
                   ^~~~~~
/home/evan/code/devpi/devpi.c:82:19: note: (near initialization for ‘pi_table[0]’)
/home/evan/code/devpi/devpi.c:83:9: error: field name not in record or union initializer
         .mode = 0644,
         ^
/home/evan/code/devpi/devpi.c:83:9: note: (near initialization for ‘pi_table[0]’)
/home/evan/code/devpi/devpi.c:83:17: warning: excess elements in scalar initializer
         .mode = 0644,
                 ^~~~
/home/evan/code/devpi/devpi.c:83:17: note: (near initialization for ‘pi_table[0]’)
/home/evan/code/devpi/devpi.c:84:9: error: field name not in record or union initializer
         .proc_handler = proc_dointvec_minmax,
         ^
/home/evan/code/devpi/devpi.c:84:9: note: (near initialization for ‘pi_table[0]’)
/home/evan/code/devpi/devpi.c:84:25: warning: excess elements in scalar initializer
         .proc_handler = proc_dointvec_minmax,
                         ^~~~~~~~~~~~~~~~~~~~
/home/evan/code/devpi/devpi.c:84:25: note: (near initialization for ‘pi_table[0]’)
/home/evan/code/devpi/devpi.c:85:9: error: field name not in record or union initializer
         .extra1 = &min_opened_threshold,
         ^
/home/evan/code/devpi/devpi.c:85:9: note: (near initialization for ‘pi_table[0]’)
/home/evan/code/devpi/devpi.c:85:19: warning: excess elements in scalar initializer
         .extra1 = &min_opened_threshold,
                   ^
/home/evan/code/devpi/devpi.c:85:19: note: (near initialization for ‘pi_table[0]’)
/home/evan/code/devpi/devpi.c:86:9: error: field name not in record or union initializer
         .extra2 = &max_opened_threshold,
         ^
/home/evan/code/devpi/devpi.c:86:9: note: (near initialization for ‘pi_table[0]’)
/home/evan/code/devpi/devpi.c:86:19: warning: excess elements in scalar initializer
         .extra2 = &max_opened_threshold,
                   ^
/home/evan/code/devpi/devpi.c:86:19: note: (near initialization for ‘pi_table[0]’)
/home/evan/code/devpi/devpi.c:88:5: warning: braces around scalar initializer
     {
     ^
/home/evan/code/devpi/devpi.c:88:5: note: (near initialization for ‘pi_table[1]’)
/home/evan/code/devpi/devpi.c:89:9: error: field name not in record or union initializer
         .procname = "max_decimal_size",
         ^
/home/evan/code/devpi/devpi.c:89:9: note: (near initialization for ‘pi_table[1]’)
/home/evan/code/devpi/devpi.c:89:21: warning: initialization makes integer from pointer without a cast [-Wint-conversion]
         .procname = "max_decimal_size",
                     ^~~~~~~~~~~~~~~~~~
/home/evan/code/devpi/devpi.c:89:21: note: (near initialization for ‘pi_table[1]’)
/home/evan/code/devpi/devpi.c:89:21: error: initializer element is not computable at load time
/home/evan/code/devpi/devpi.c:89:21: note: (near initialization for ‘pi_table[1]’)
/home/evan/code/devpi/devpi.c:90:9: error: field name not in record or union initializer
         .data = &max_decimal_size,
         ^
/home/evan/code/devpi/devpi.c:90:9: note: (near initialization for ‘pi_table[1]’)
/home/evan/code/devpi/devpi.c:90:17: warning: excess elements in scalar initializer
         .data = &max_decimal_size,
                 ^
/home/evan/code/devpi/devpi.c:90:17: note: (near initialization for ‘pi_table[1]’)
/home/evan/code/devpi/devpi.c:91:9: error: field name not in record or union initializer
         .maxlen = sizeof(int),
         ^
/home/evan/code/devpi/devpi.c:91:9: note: (near initialization for ‘pi_table[1]’)
/home/evan/code/devpi/devpi.c:91:19: warning: excess elements in scalar initializer
         .maxlen = sizeof(int),
                   ^~~~~~
/home/evan/code/devpi/devpi.c:91:19: note: (near initialization for ‘pi_table[1]’)
/home/evan/code/devpi/devpi.c:92:9: error: field name not in record or union initializer
         .mode = 0644,
         ^
/home/evan/code/devpi/devpi.c:92:9: note: (near initialization for ‘pi_table[1]’)
/home/evan/code/devpi/devpi.c:92:17: warning: excess elements in scalar initializer
         .mode = 0644,
                 ^~~~
/home/evan/code/devpi/devpi.c:92:17: note: (near initialization for ‘pi_table[1]’)
/home/evan/code/devpi/devpi.c:93:9: error: field name not in record or union initializer
         .proc_handler = proc_dointvec,
         ^
/home/evan/code/devpi/devpi.c:93:9: note: (near initialization for ‘pi_table[1]’)
/home/evan/code/devpi/devpi.c:93:25: warning: excess elements in scalar initializer
         .proc_handler = proc_dointvec,
                         ^~~~~~~~~~~~~
/home/evan/code/devpi/devpi.c:93:25: note: (near initialization for ‘pi_table[1]’)
/home/evan/code/devpi/devpi.c:95:5: warning: braces around scalar initializer
     {
     ^
/home/evan/code/devpi/devpi.c:95:5: note: (near initialization for ‘pi_table[2]’)
/home/evan/code/devpi/devpi.c:96:9: error: field name not in record or union initializer
         .procname = "mode",
         ^
/home/evan/code/devpi/devpi.c:96:9: note: (near initialization for ‘pi_table[2]’)
/home/evan/code/devpi/devpi.c:96:21: warning: initialization makes integer from pointer without a cast [-Wint-conversion]
         .procname = "mode",
                     ^~~~~~
/home/evan/code/devpi/devpi.c:96:21: note: (near initialization for ‘pi_table[2]’)
/home/evan/code/devpi/devpi.c:96:21: error: initializer element is not computable at load time
/home/evan/code/devpi/devpi.c:96:21: note: (near initialization for ‘pi_table[2]’)
/home/evan/code/devpi/devpi.c:97:9: error: field name not in record or union initializer
         .data = &current_mode,
         ^
/home/evan/code/devpi/devpi.c:97:9: note: (near initialization for ‘pi_table[2]’)
/home/evan/code/devpi/devpi.c:97:17: warning: excess elements in scalar initializer
         .data = &current_mode,
                 ^
/home/evan/code/devpi/devpi.c:97:17: note: (near initialization for ‘pi_table[2]’)
/home/evan/code/devpi/devpi.c:98:9: error: field name not in record or union initializer
         .proc_handler = proc_dopimode,
         ^
/home/evan/code/devpi/devpi.c:98:9: note: (near initialization for ‘pi_table[2]’)
/home/evan/code/devpi/devpi.c:98:25: error: ‘proc_dopimode’ undeclared here (not in a function)
         .proc_handler = proc_dopimode,
                         ^~~~~~~~~~~~~
/home/evan/code/devpi/devpi.c:98:25: warning: excess elements in scalar initializer
/home/evan/code/devpi/devpi.c:98:25: note: (near initialization for ‘pi_table[2]’)
/home/evan/code/devpi/devpi.c:99:9: error: field name not in record or union initializer
         .mode = 0644,
         ^
/home/evan/code/devpi/devpi.c:99:9: note: (near initialization for ‘pi_table[2]’)
/home/evan/code/devpi/devpi.c:99:17: warning: excess elements in scalar initializer
         .mode = 0644,
                 ^~~~
/home/evan/code/devpi/devpi.c:99:17: note: (near initialization for ‘pi_table[2]’)
/home/evan/code/devpi/devpi.c:101:5: warning: braces around scalar initializer
     { }
     ^
/home/evan/code/devpi/devpi.c:101:5: note: (near initialization for ‘pi_table[3]’)
/home/evan/code/devpi/devpi.c:103:8: error: unknown type name ‘ctl_table’
 static ctl_table pi_root[] = {
        ^~~~~~~~~
/home/evan/code/devpi/devpi.c:104:5: warning: braces around scalar initializer
     {
     ^
/home/evan/code/devpi/devpi.c:104:5: note: (near initialization for ‘pi_root[0]’)
/home/evan/code/devpi/devpi.c:105:9: error: field name not in record or union initializer
         .procname = "pi",
         ^
/home/evan/code/devpi/devpi.c:105:9: note: (near initialization for ‘pi_root[0]’)
/home/evan/code/devpi/devpi.c:105:21: warning: initialization makes integer from pointer without a cast [-Wint-conversion]
         .procname = "pi",
                     ^~~~
/home/evan/code/devpi/devpi.c:105:21: note: (near initialization for ‘pi_root[0]’)
/home/evan/code/devpi/devpi.c:105:21: error: initializer element is not computable at load time
/home/evan/code/devpi/devpi.c:105:21: note: (near initialization for ‘pi_root[0]’)
/home/evan/code/devpi/devpi.c:106:9: error: field name not in record or union initializer
         .mode = 0555,
         ^
/home/evan/code/devpi/devpi.c:106:9: note: (near initialization for ‘pi_root[0]’)
/home/evan/code/devpi/devpi.c:106:17: warning: excess elements in scalar initializer
         .mode = 0555,
                 ^~~~
/home/evan/code/devpi/devpi.c:106:17: note: (near initialization for ‘pi_root[0]’)
/home/evan/code/devpi/devpi.c:107:9: error: field name not in record or union initializer
         .child = pi_table,
         ^
/home/evan/code/devpi/devpi.c:107:9: note: (near initialization for ‘pi_root[0]’)
/home/evan/code/devpi/devpi.c:107:18: warning: excess elements in scalar initializer
         .child = pi_table,
                  ^~~~~~~~
/home/evan/code/devpi/devpi.c:107:18: note: (near initialization for ‘pi_root[0]’)
/home/evan/code/devpi/devpi.c:109:5: warning: braces around scalar initializer
     { }
     ^
/home/evan/code/devpi/devpi.c:109:5: note: (near initialization for ‘pi_root[1]’)
/home/evan/code/devpi/devpi.c:111:8: error: unknown type name ‘ctl_table’
 static ctl_table dev_root[] = {
        ^~~~~~~~~
/home/evan/code/devpi/devpi.c:112:5: warning: braces around scalar initializer
     {
     ^
/home/evan/code/devpi/devpi.c:112:5: note: (near initialization for ‘dev_root[0]’)
/home/evan/code/devpi/devpi.c:113:9: error: field name not in record or union initializer
         .procname = "dev",
         ^
/home/evan/code/devpi/devpi.c:113:9: note: (near initialization for ‘dev_root[0]’)
/home/evan/code/devpi/devpi.c:113:21: warning: initialization makes integer from pointer without a cast [-Wint-conversion]
         .procname = "dev",
                     ^~~~~
/home/evan/code/devpi/devpi.c:113:21: note: (near initialization for ‘dev_root[0]’)
/home/evan/code/devpi/devpi.c:113:21: error: initializer element is not computable at load time
/home/evan/code/devpi/devpi.c:113:21: note: (near initialization for ‘dev_root[0]’)
/home/evan/code/devpi/devpi.c:114:9: error: field name not in record or union initializer
         .mode = 0555,
         ^
/home/evan/code/devpi/devpi.c:114:9: note: (near initialization for ‘dev_root[0]’)
/home/evan/code/devpi/devpi.c:114:17: warning: excess elements in scalar initializer
         .mode = 0555,
                 ^~~~
/home/evan/code/devpi/devpi.c:114:17: note: (near initialization for ‘dev_root[0]’)
/home/evan/code/devpi/devpi.c:115:9: error: field name not in record or union initializer
         .child = pi_root,
         ^
/home/evan/code/devpi/devpi.c:115:9: note: (near initialization for ‘dev_root[0]’)
/home/evan/code/devpi/devpi.c:115:18: warning: excess elements in scalar initializer
         .child = pi_root,
                  ^~~~~~~
/home/evan/code/devpi/devpi.c:115:18: note: (near initialization for ‘dev_root[0]’)
/home/evan/code/devpi/devpi.c:117:5: warning: braces around scalar initializer
     { }
     ^
/home/evan/code/devpi/devpi.c:117:5: note: (near initialization for ‘dev_root[1]’)
/home/evan/code/devpi/devpi.c: In function ‘init_sysctl’:
/home/evan/code/devpi/devpi.c:122:44: error: passing argument 1 of ‘register_sysctl_table’ from incompatible pointer type [-Werror=incompatible-pointer-types]
     sysctl_handler = register_sysctl_table(dev_root);
                                            ^~~~~~~~
In file included from ./include/linux/timer.h:258:0,
                 from ./include/linux/workqueue.h:8,
                 from ./include/linux/srcu.h:34,
                 from ./include/linux/notifier.h:15,
                 from ./include/linux/memory_hotplug.h:6,
                 from ./include/linux/mmzone.h:751,
                 from ./include/linux/gfp.h:5,
                 from ./include/linux/kmod.h:22,
                 from ./include/linux/module.h:13,
                 from /home/evan/code/devpi/devpi.c:6:
./include/linux/sysctl.h:190:26: note: expected ‘struct ctl_table *’ but argument is of type ‘int *’
 struct ctl_table_header *register_sysctl_table(struct ctl_table * table);
                          ^~~~~~~~~~~~~~~~~~~~~
cc1: some warnings being treated as errors
scripts/Makefile.build:299: recipe for target '/home/evan/code/devpi/devpi.o' failed
make[2]: *** [/home/evan/code/devpi/devpi.o] Error 1
Makefile:1494: recipe for target '_module_/home/evan/code/devpi' failed
make[1]: *** [_module_/home/evan/code/devpi] Error 2
make[1]: Leaving directory '/usr/src/kernels/4.9.13-201.fc25.x86_64'
Makefile:14: recipe for target 'all' failed
make: *** [all] Error 2
```

--- a/devpi.c
+++ b/devpi.c
@@ -41,9 +41,9 @@ static unsigned min_opened_threshold = 1;
 static unsigned max_opened_threshold = 2;
 
 /* Read the string equivalents of the enum pi_mode modes */
-static int proc_dopimode(ctl_table* table, int write, void __user* buffer, size_t* lenp, loff_t* ppos)
+static int proc_dopimode(struct ctl_table* table, int write, void __user* buffer, size_t* lenp, loff_t* ppos)
 {
-    ctl_table fake_table;
+    struct ctl_table fake_table;
     char buf[10];
     enum pi_mode pmode;
 
@@ -75,7 +75,7 @@ static int proc_dopimode(ctl_table* table, int write, void __user* buffer, size_
     }
 }
 
-static ctl_table pi_table[] = {
+static struct ctl_table pi_table[] = {
     {
         .procname = "max_opened",
         .data = &max_opened,
@@ -100,7 +100,7 @@ static ctl_table pi_table[] = {
     },
     { }
 };
-static ctl_table pi_root[] = {
+static struct ctl_table pi_root[] = {
     {
         .procname = "pi",
         .mode = 0555,
@@ -108,7 +108,7 @@ static ctl_table pi_root[] = {
     },
     { }
 };
-static ctl_table dev_root[] = {
+static struct ctl_table dev_root[] = {
     {
         .procname = "dev",
         .mode = 0555,

